vue/components/atoms/error-alert/error-alert.vue

```diff
@@ -0,0 +1,298 @@
+    opacity: 1;
+}
+
+.error-alert > .error-container > .error-stack {
+    background-color: #e96760;
+    max-height: 300px;
+    overflow-x: auto;
+    overflow-y: auto;
+}
+
+.error-alert > .error-container > .error-stack > .stack-line {
+    margin: 0px;
+    padding: 5px 8px 5px 26px;
+}
+
+.error-alert > .error-container > .error-stack > .stack-line:nth-child(odd) {
+    background-color: #ec7a74;
+}
+
+.error-alert.fade-enter-active,
+.error-alert.fade-leave-active {
+    transition: opacity 0.2s ease-in-out;
+}
+
+.error-alert.fade-enter,
+.error-alert.fade-leave-to {
+    opacity: 0;
+}
+
+.error-alert > .error-container > .error-stack.fade-enter-active,
+.error-alert > .error-container > .error-stack.fade-leave-active {
+    transition: max-height 0.2s ease-in-out, opacity 0.2s ease-in-out;
+}
+
+.error-alert > .error-container > .error-stack.fade-enter,
+.error-alert > .error-container > .error-stack.fade-leave-to {
+    max-height: 0px;
+    opacity: 0.5;
+}
+
+body.mobile .error-alert > .error-container > .error-header > .right > .arrow,
+body.tablet .error-alert > .error-container > .error-header > .right > .arrow {
+    opacity: 1;
+}
+
+body.mobile .error-alert > .error-container > .error-header > .right > .close,
+body.tablet .error-alert > .error-container > .error-header > .right > .close {
+    opacity: 1;
+}
+</style>
+
+<script>
+export const ErrorAlert = {
+    name: "error-alert",
+    props: {
+        errorObject: {
+            type: Error,
+            required: true
+        },
+        visible: {
+            type: Boolean,
+            default: true
+        },
+        collapsed: {
+            type: Boolean,
+            default: true
+        },
+        direction: {
+            type: String,
+            default: "bottom"
+        }
+    },
+    computed: {
+        classes() {
+            const base = {};
+            base.collapsed = this.isCollapsed;
+            if (this.direction) {
+                base[`direction-${this.direction}`] = true;
+            }
+            return base;
+        },
+        arrowIcon() {
+            if (this.isCollapsed) {
+                return "chevron-down";
+            } else {
+                return "chevron-up";
+            }
+        },
+        isVisible() {
+            return this.visibleData;
+        },
+        isCollapsed() {
+            return this.collapsedData;
+        },
+        message() {
+            return this.errorObject.message;
+        },
+        file() {
+            return this.errorObject.fileName;
+        },
+        sourceLocation() {
+            if (this.errorObject.lineNumber === undefined) return;
+            if (this.errorObject.columnNumber === undefined) return;
+            return `${this.errorObject.lineNumber}:${this.errorObject.columnNumber}`;
+        },
+        stackLines() {
+            return this.errorObject.stack.split("\n").filter(l => l);

```

Comment:
this is to filter through empty lines? Perhaps add a comment so that it becomes explicit or modify to `return this.errorObject.stack.split("\n").filter(l => l !== "");`, whatever you think it's best